nexus/resource_repository_dockerhostedwithoutports_test.go

```diff
@@ -11,20 +10,56 @@ import (
 
 func TestAccRepositoryDockerHostedWithoutPorts(t *testing.T) {
 	repoName := fmt.Sprintf("test-repo-docker-hosted-%s", acctest.RandString(10))
-	repoOnline := true
 
 	resource.Test(t, resource.TestCase{
 		PreCheck:  func() { testAccPreCheck(t) },
 		Providers: testAccProviders,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccRepositoryDockerHostedWithoutPorts(repoName, repoOnline),
-				// TODO: add check for storage
-				// TODO: add check for repository connectors
-				// TODO: add check for Group members
-				// TODO: add check for api version support
-				// TODO: add tests for readonly repository
-				// TODO: add tests for cleanup
+				Config: createTfStmtForResourceDockerHostedWithPorts(repoName),
+				Check: resource.ComposeTestCheckFunc(
+					// Base and common repo props
+					// Identity fields
+					resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "id", repoName),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "name", repoName),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "format", "docker"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "type", "hosted"),
+					),
+					// Common fields
+					// Online
+					resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "online", "true"),
+						// Storage
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "storage.#", "1"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "storage.0.blob_store_name", "default"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "storage.0.strict_content_type_validation", "true"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "storage.0.write_policy", "ALLOW"),
+					),
+					// No fields related to other repo types
+					// Format
+					resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "apt.#", "0"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "apt_signing.#", "0"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "bower.#", "0"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "maven.#", "0"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "docker_proxy.#", "0"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "http_client.#", "0"),
+					),
+					// Type
+					resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "proxy.#", "0"),
+						resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "negative_cache.#", "0"),
+					),
+
+					// Fields related to this format and type
+					// Format
+					resource.ComposeAggregateTestCheckFunc(
```

Comment:
Please add `resource.TestCheckResourceAttr("nexus_repository.docker_hosted", "docker.0.http_port", "0"),` to check that this field exist and will be stored as 0 by default